storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -286,12 +293,23 @@ func (s *peersSource) incrementalFlush(
 				break
 			}
 
-			err = prepared.Persist(series.ID, segment, bl.Checksum())
+			err = prepared.Persist(s.ID, segment, bl.Checksum())
 			tmpCtx.BlockingClose()
 			if err != nil {
 				blockErr = err // Need to call prepared.Close, avoid return
 				break
 			}
+
+			switch seriesCachePolicy {
+			case series.CacheAll:
+			case series.CacheAllMetadata:
```

Comment:
So for CacheAllMetadata we want to retain the blocks but remove the data from memory.  For the RecentlyRead we want to remove the block completely, because we don't need it in memory at all.  

The one on the reader is completely independent to this.

Reply:
that would be dope